search/search_repository.go

```diff
@@ -430,7 +444,10 @@ func parseFilterString(ctx context.Context, rawSearchString string) (criteria.Ex
 	q := Query{}
 	parseMap(fm, &q)
 
-	result := q.generateExpression()
+	result, err := q.generateExpression()
+	if err != nil {
```

Comment:
Sorry for being picky but `IF` block is not needed we can just `return res, err` and the caller can verify if its an error or not.